### PR TITLE
Migrating Lucene.Net.Core

### PR DESCRIPTION
**Pull request by conniey**

### CODE REVIEW

File: src/Lucene.Net.Core/Store/NativeFSLockFactory.cs [hunk 1]
```diff
@@ -205,9 +203,6 @@ public override void Release()
                 {
                     try
                     {
-                        //TODO: conniey
-                        //Channel.Unlock(0, Channel.Length);
```
Comment (EMaher) [inline src/Lucene.Net.Core/Store/NativeFSLockFactory.cs]: Why is this removed rather than if-def'ed.

File: src/Lucene.Net.Core/Support/Configuration/Configuration.cs [hunk 2]
```diff
@@ -1,28 +1,34 @@
...
-namespace Lucene.Net.TestFramework.Support
+namespace Lucene.Net.Support.Configuration
 {
-    public static class SystemProperties
+    public static class Configuration
     {
-        public static string GetProperty(string key)
+        private static IConfigurationRoot _configuration;
+
+        static Configuration()
```
Comment (EMaher) [inline src/Lucene.Net.Core/Support/Configuration/Configuration.cs]: Doesn't this need to be if-def'ed?  I though Microsoft.Extensions.Configuration classes only exist in NET CORE.

File: src/Lucene.Net.Core/Store/NativeFSLockFactory.cs (hunk 1 above)
Comment (conniey) [inline src/Lucene.Net.Core/Store/NativeFSLockFactory.cs]: It was unnecessary. There is a check before it that ensures this we don't try this statement twice.
[NativeFSLockFactory.cs: L139](https://github.com/conniey/lucenenet/blob/3518f32be4e744074ea7bbb429a11f31d968bc1e/src/Lucene.Net.Core/Store/NativeFSLockFactory.cs#L139)

File: src/Lucene.Net.Tests/Lucene.Net.Tests.Portable.csproj [hunk 3]
```diff
@@ -5,12 +5,9 @@
     <MinimumVisualStudioVersion>14.0</MinimumVisualStudioVersion>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{DBC3B677-805C-430E-AF78-A9ABF3975D17}</ProjectGuid>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <Platform Condition=" '$(Platform)' == '' ">x86</Platform>
     <ProjectGuid>{91AD3FB7-8BC0-4008-852B-9972CD47E6C0}</ProjectGuid>
     <OutputType>Library</OutputType>
-    <AssemblyName>Lucene.Net.TestsPortable</AssemblyName>
+    <AssemblyName>Lucene.Net.Tests.Portable</AssemblyName>
```
Comment (EMaher) [inline src/Lucene.Net.Tests/Lucene.Net.Tests.Portable.csproj]: I thought we were talking 'Portable' out of the assembly name.

Comment (conniey) [inline src/Lucene.Net.Tests/Lucene.Net.Tests.Portable.csproj]: Only problem is that if they try to build the desktop solution and then the portable solution.. it'll overwrite the same DLL because they'll share the same assembly name.

**Comment by conniey** [comment]
I fixed the desktop solution so it builds as well.

**Comment by EMaher** [comment]
Excellent.  That was my main concern.  I'll merge this PR.

File: src/Lucene.Net.Tests/Lucene.Net.Tests.Portable.csproj (hunk 3 above)
Comment (EMaher) [inline src/Lucene.Net.Tests/Lucene.Net.Tests.Portable.csproj]: No objections, I just thought I saw a previous change where the 'Portable' was taken out.  I look through the project files and make sure that isn't the case.
